unfurl/job.py

```diff
@@ -780,7 +791,22 @@ def apply(
             elif isinstance(taskRequest, TaskRequestGroup):
                 _task = self.apply_group(depth, taskRequest)
             else:
-                _task = self._run_operation(taskRequest, workflow, depth)
+                ci = os.environ.get("CI", False) # Running in a CI environment (eg GitLab CI)
+                if ci:
+                    # Start collapsible section
+                    # https://docs.gitlab.com/ee/ci/jobs/#expand-and-collapse-job-log-sections
+                    # The octal \033 is used instead of \e because python doesn't recognize \e as an escape sequence
+                    print(f"\033[0Ksection_start:{int(time.time())}:task_{hash(taskRequest)}[collapsed=true]\r\033[0KTask {taskRequest.target.name} ({taskRequest}")
+
+                _task = self._run_operation(taskRequest, workflow, depth, meta={
+                    "task_number": idx,
+                    "task_count": len(taskRequests),
+                    "logger": logging.getLogger(f"unfurljob.{idx}")
```

Comment:
I think we want to hide the json output from the user.. What if we setup the formatter for this logger to use the same "\033[0K" console escape code trick that the section output uses to hide the output from the user.

Reply:
Done, had to add a condition in `ColorHandler` to hide output because it appears to get handled by both it and the custom job handler